### ISSUE TITLE
Documentation how to deal with `ResourceException`

### ISSUE THREAD

**Issue by EarthCitizen** [okta/okta-sdk-java#122]
Many or all of the methods on Client throw exceptions under various circumstances, but this fact is not documented at all. There needs to be documentation for all exceptions thrown so that consumers of this API can know what to handle under which circumstances and know what to do to have a stable application.

```
getUser
@Generated(value="com.okta.swagger.codegen.OktaJavaClientApiCodegen",
           date="2017-09-05T20:10:20.883-04:00",
           comments="GET - /api/v1/users/{userId}")
User getUser(String userId)
Parameters:
userId - (required)
Returns:
User
```

**Comment by bdemers** [comment]
Can you give a specific example?  Are you talking about the Runtime Exceptions (i.e. `ResourceException`)?  Or something else?  We are planning on adding a doc page for Error handling (but this wouldn't be reflected in the javadoc / method signature)

**Comment by EarthCitizen** [comment]
The example you gave is what I am referring to.

**Comment by bdemers** [comment]
@EarthCitizen I appreciate the snippet, but I'm still not 100% what you are looking for.  Are you looking for a list of RuntimeExceptions that could be thrown, (if that is the case `ResourceException` is the only one explicitly thrown)

**Comment by EarthCitizen** [comment]
Currently, there is no contract specified in the JavaDocs for the behavior of the API in relation to errors. Therefore, the only thing that can be known is that if an exception occurs, it will be some kind of RuntimeException because there are no checked exceptions. That is all that can be discerned by reading the docs. So, a consumer of the API has no choice but to catch RuntimeException, which is bad practice, because the documentation does not provide a behavior contract.

**Comment by EarthCitizen** [comment]
If `ResourceException` is the only way to intercept and respond to an error state, that should be documented.

**Comment by bdemers** [comment]
@EarthCitizen can do!

**Comment by EarthCitizen** [comment]
Related issue here: #111

**Comment by bdemers** [comment]
Fixed in 0.9.0
